linera-execution/src/runtime.rs

```diff
@@ -1361,14 +1364,22 @@ impl ContractRuntime for ContractSyncRuntimeHandle {
             application_id: application_id.into(),
         };
         let subscriber_app_id = this.current_application().id;
-        this.execution_state_sender
+        let next_index = this
+            .execution_state_sender
```

Comment:
It's really unfortunate that we need to communicate via async channels to find out the next index of the event. Seems like an overkill and potential source of bugs (thing like this should. be atomic and single-threaded).

Reply:
I agree! This execution state actor vs. runtime split is making a lot of things awkward. I want to look into this soon, but it'll be quite a large change (if it's feasible to resolve that at all).